client.go

```diff
@@ -37,7 +41,9 @@ func (client Client) Call(method string, params interface{}, results interface{}
 		fmt.Printf("REQUEST: %s\n", string(reqBytes))
 	}
 	req, err := http.NewRequest("POST", client.addr, bytes.NewBuffer(reqBytes))
-
+	if client.username != nil {
```

Comment:
this is never going to be nil, not even sure if it will build should compare to != ""